mycroft/tts/festival_tts.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2020 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import subprocess
+
+from .tts import TTS, TTSValidator
+
+
+class Festival(TTS):
+    def __init__(self, lang, config):
+        super(Festival, self).__init__(lang, config, FestivalValidator(self))
+
+    def execute(self, sentence, ident=None, listen=False):
+
+        encoding = self.config.get('encoding')
+        cmd = "echo \"" + sentence + "\""
+
+        if encoding != 'utf8':
+            cmd += " | iconv -f utf8 -t " + encoding
+
+        cmd += " | festival --tts --language " + self.lang
+
+        self.begin_audio()
+        subprocess.call(cmd, shell=True)
```

Comment:
`shell=True` has some [security considerations](https://docs.python.org/3/library/subprocess.html#security-considerations) so ideally we can avoid using it. Unfortunately you can't use a simple pipe character without it.

One option would be to split each command into it's own call and pipe using subprocesses built in method. Eg:
```python
        text = subprocess.Popen(('echo', sentence), stdout=subprocess.PIPE)

        if encoding != 'utf8':
            convert_cmd = ('iconv', '-f', 'utf8', '-t', encoding)
            converted_text = subprocess.Popen(convert_cmd, 
                                              stdin=text.stdout,
                                              stout=subprocess.PIPE)
            text.wait()
            text = converted_text

        tts_cmd = ('festival', '--tts', '--language', lang)
        self.begin_audio()
        subprocess.call(tts_cmd, stdin=text.stdout)
        text.wait()
        self.end_audio(listen)
```